docs/guide-definitions.rst

```diff
@@ -7,7 +7,7 @@ Interval:
     - An *interval* is a tuple of integers (start, end) with start <= end.  
     - Coordinates are assumed to be 0-based and intervals half-open (1-based ends) i.e. [start, end). 
     - An interval has a *length* equal to (end - start). 
-    - A special case where start and end are the same, i.e. [X, X), is interpreted as a *point* (aka an *empty interval*, i.e. an edge between 1-bp bins). A point has zero length. 
+    - A special case where start and end are the same, i.e. [X, X), is interpreted as a *point* (aka an *empty interval*, i.e. an edge between 1-bp bins). A point has zero length. For consistency of the operations such as subtract, overlap, etc. point intervals are treated as len1 segments. 
```

Comment:
I would not add this additional statement, as this is just the internal behavior and does not change the definition per se.

Reply:
I still think the definition may be better w/o the last statement, as that is an internal "virtual conversion" to the len1 segment which may be mis-understood.

If we need more explanation, how about the following? cc @nvictus @golobor  

  - A special case where start and end are the same, i.e. [X, X), is interpreted as a *point* (aka an *empty interval*, i.e. an edge between 1-bp bins). The point [X,X) overlaps interval [X,X+1), but does not overlap the interval [X-1,X).